packages/base-contract/src/utils.ts

```diff
@@ -0,0 +1,25 @@
+import { DataItem } from '@0xproject/types';
+import * as _ from 'lodash';
+
+// tslint:disable-next-line:completed-docs
+export function formatABIDataItem(abi: DataItem, value: any, formatter: (type: string, value: any) => any): any {
+    const trailingArrayRegex = /\[\d*\]$/;
+    if (abi.type.match(trailingArrayRegex)) {
+        const arrayItemType = abi.type.replace(trailingArrayRegex, '');
+        return _.map(value, val => {
+            const arrayItemAbi = {
+                ...abi,
+                type: arrayItemType,
+            };
+            return formatABIDataItem(arrayItemAbi, val, formatter);
+        });
+    } else if (abi.type === 'tuple') {
```

Comment:
This is not the abi type, but the data type. We don't have a type for that. This will be the list of all solidity types which is infinite. `int`, `int[]`, `int[][]` etc.